registry/auth/token/accesscontroller.go

```diff
@@ -134,23 +160,31 @@ func (ac authChallenge) SetHeaders(r *http.Request, w http.ResponseWriter) {
 
 // accessController implements the auth.AccessController interface.
 type accessController struct {
-	realm        string
-	autoRedirect bool
-	issuer       string
-	service      string
-	rootCerts    *x509.CertPool
-	trustedKeys  map[string]crypto.PublicKey
+	realm                string
+	autoRedirect         bool
+	autoRedirectPath     string
+	autoRedirectForceTLS bool
+	issuer               string
+	service              string
+	rootCerts            *x509.CertPool
+	trustedKeys          map[string]crypto.PublicKey
 }
 
+const (
+	defaultAutoRedirectPath = "/auth/token"
+)
+
 // tokenAccessOptions is a convenience type for handling
 // options to the contstructor of an accessController.
 type tokenAccessOptions struct {
-	realm          string
-	autoRedirect   bool
-	issuer         string
-	service        string
-	rootCertBundle string
-	jwks           string
+	realm                        string
+	autoRedirect                 bool
+	autoRedirectPath             string
+	autoRedirectForceTLSDisabled bool
```

Comment:
If we are going to be adding new params we need to update the docs before this gets merged.

As an aside, I'm not sure `autoRedirectForceTLSDisabled` is required here as whether TLS is used or not is implied by [protocol] `scheme`  🤷‍♂️

Reply:
OK, I'll update the docs.

I'm not sure too, we need this if we want the origin behavior that force to return https.